superset/views/core.py

```diff
@@ -1363,7 +1363,9 @@ def testconn(self):
         except Exception as e:
             logging.exception(e)
             return json_error_response(
-                "Connection failed!\n\n" "The error message returned was:\n{}".format(e)
+                "Connection failed!\n\n"
+                "The error message returned was:\n{}".format(e),
```

Comment:
While logging is usually done with `format` to save cycles, in this instance I believe using a regular f-string is the correct course of action.

Reply:
Thanks for the review Ville! I'm happy to use either approach. As I'm still somewhat new to Python, it would be very useful for me to understand the reasoning behind your suggestion. Would you mind expanding a little bit?

Reply:
As Superset moved to Python 3.6+, we now generally prefer to use f-strings over `"".format()` whenever possible. This is mostly due to f-strings being more readable, but they are usually also slightly more performant in normal usage. While the codebase is still filled with `format()`,when touching old code the convention has been to change them to f-strings. An exception to this rule is logging, which `pylint` (I believe) checks for, for which one should generally use `logger.debug('message: %s', c)` style. Check https://docs.python.org/3/howto/logging.html#optimization

Hope this helps!

Reply:
Yes, that clarifies it. Thank you!

Reply:
Hmm...in this case @villebro the line number you honed in on is returning an error response, not a logger call. I think that means `"".format()` is appropriate following the logic you posted?

Reply:
Sorry, I might have been unclear in my comment. What I meant to say was that in the above case I think using an f-string should be ok, and that f-strings should mainly be avoided in logging messages. The same applies to translations, in which variable text should not be burned into the message prior to translation.

Reply:
OK @villebro I'm confused. What change precisely are you looking for on this PR?

Reply:
@willbarrett in my original comment I was pointing out when one should and should not use f-strings. In the case above I would personally write
```python
            return json_error_response(
                "Connection failed!\n\n"
                f"The error message returned was:\n{e}",
                400,
            )
```

Reply:
updated